Archospring victory

keyz182 opened the pull request:
#7 

Still a work in progress, don't merge.

Looked to be easier to reimplement that to try to patch the archo quest. 

ToDo:
~1) double check the storyteller patch to make sure we're making this quest auto-triggering~
~2) Validate it works - placing a Power Regulator should prompt the first quest.~
~3) Descriptions and text - all still archonexus. Maybe for moving maps something like "As the machine roars into action, the ground seems to become unstable. You only have a few moments to grab your things and go".~
~4) Set up "the victory".~
~5) Create new `Screen_ArchonexusSettlementCinematics`, subclass `QuestPart_NewColony` hook them together, then replace `QuestPart_NewColony` in `QuestNode_Root_WaterVictory_Cycle` with the new one.~

keyz182 committed:
First Pass at Water victory.

keyz182 committed:
"Working" in so far as the quest triggers work.

keyz182 committed:
New base title screen

keyz182 committed:
Fixup project structure

keyz182 committed:
Facilitate custom site gen

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_WaterVictory_Cycle.cs:
```diff
@@ -0,0 +1,112 @@
+      slate.Set("playerSettlementsCount", var.Count);
+      slate.Set("colonistsAllowed", 5);
+      slate.Set("animalsAllowed", 5);
+      slate.Set("requiredWealth", 350000f);
```
presumably not relevant

keyz182 replied:
Nope, will do a cleanup pass for all the gumph later today.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_WaterVictory_Cycle.cs:
```diff
@@ -0,0 +1,112 @@
+      }
+      slate.Set("playerSettlements", var);
+      slate.Set("playerSettlementsCount", var.Count);
+      slate.Set("colonistsAllowed", 5);
+      slate.Set("animalsAllowed", 5);
```
should probably be higher, we're not trying to restrict in the same way right he should be able to take more people.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_WaterVictory_Cycle.cs:
```diff
@@ -0,0 +1,112 @@
+    {
+        Quest quest = QuestGen.quest;
+        Slate slate = QuestGen.slate;
+        string newSignal1 = QuestGen.GenerateNewSignal("NewColonyCreated");
```
I know it's copy pasta but I'd dearly love the signals to have sensible names so it's more copiable and grokkable

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_WaterVictory_Cycle.cs:
```diff
@@ -0,0 +1,112 @@
+      QuestGen.GenerateNewSignal("ActivateLetterReminderSignal");
+      quest.AddPart(Requirement);
+
+      QuestPartActivable part1 = Part1;
+      part1.inSignalEnable = quest.AddedSignal;
+      part1.outSignalsCompleted.Add(QuestSignal);
+      part1.signalListenMode = QuestPart.SignalListenMode.NotYetAcceptedOnly;
+      quest.AddPart(part1);
+
+      QuestPart_PassOutInterval part2 = new QuestPart_PassOutInterval();
+      part2.signalListenMode = QuestPart.SignalListenMode.NotYetAcceptedOnly;
+      part2.inSignalEnable = QuestSignal;
+      part2.ticksInterval = new IntRange(3600000, 3600000);
+      part2.outSignals.Add(newSignal2);
+      quest.AddPart(part2);
+
+      QuestPart_Filter part3 = Part3;
+      part3.inSignal = newSignal2;
+      part3.outSignal = QuestGen.GenerateNewSignal("OuterNodeCompleted");
+      part3.signalListenMode = QuestPart.SignalListenMode.NotYetAcceptedOnly;
+      quest.AddPart(part3);
```
I simply can't parse this, I know it's copy paste but part one is an activatable quest part that is completed when built > 3; fine, but part three is a filter that checks the same condition for reasons? I assume it works but this is where I throw my hands up and say I don't understand the quest system

keyz182 replied:
Yeah, copypasta, so went with it. I'll have a play later to see if it's redundant or actually needed.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_CatalyticSeparator.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public override void QuestPartTick()
+    {
+        if(Find.TickManager.TicksGame % 60 != 0 || WaterVictoryWorldComponent.Instance.CatalyticSeparatorsBuilt < 3)
```
the targets shouldn't be repeatedly hard coded in so many places, grab them from settings or a static or something

feldoh reviewed (commented):
honestly shrug, quests are basically unreviewable, part1 part3 signals that come from ? 🤷

keyz182 committed:
Mostly working, needs cleanup

keyz182 committed:
Finish off the quest titles and descriptions. Clean up code and dialogs.

keyz182 committed:
Map gen experiments

keyz182 committed:
Workspace update

keyz182 committed:
Map gen in a good place

keyz182 committed:
rebuild

keyz182 committed:
Drop old DLLs

keyz182 committed:
More map gen, clean up, replace xml patch with harmony patch (because xml patch stopped working)

keyz182 committed:
Experiment with relic style subquests. Working, but the e2e quest chain isn't now until I fully integrate

keyz182 committed:
Working pre-quest

keyz182 committed:
Fix Settings.cs

keyz182 committed:
Comments

keyz182 committed:
Pre-quest

keyz182 committed:
Components haulable to buildings

keyz182 committed:
More work/cleanup

keyz182 committed:
Nearly e2e

keyz182 committed:
rebuild

keyz182 committed:
Start documenting as a sanity pass

keyz182 committed:
Fully working final quest. Still needs some variety in random component quests, but otherwise good.

keyz182 committed:
Trade requests and prevent sites despawning

feldoh commented on 1.5/Defs/BuildingDefs/MSS_Thirst_WaterQuest.xml:
```diff
@@ -0,0 +1,84 @@
+<Defs>
+
+
+    <ThingDef Name="MSS_ArchoQuestBuilding" ParentName="BuildingBase" Abstract="True">
```
Missing _Thirst namespacing; I'm doing a full pass on that to bring it inline though there's a few that it's too late for

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/HarmonyPatches/Site_Patch.cs:
```diff
@@ -0,0 +1,30 @@
+[HarmonyPatch(typeof(Site))]
+public static class Site_Patch
+{
+    public static List<SitePartDef> SitesToNotDespawn = [
```
HashSet not list for membership

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/HarmonyPatches/StorytellerComp_RefiringUniqueQuest_Patch.cs:
```diff
@@ -0,0 +1,61 @@
+            else
+            {
+                if (!(bool) generateSkipped.Value.GetValue(__instance)
+                        ? IntervalsPassed == (int) (Props.minDaysPassed * 60.0) + 1
```
same again, I think this should be >=
```suggestion
                        ? IntervalsPassed >= (int) (Props.minDaysPassed * 60.0) + 1
```

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/HarmonyPatches/StorytellerComp_RefiringUniqueQuest_Patch.cs:
```diff
@@ -0,0 +1,61 @@
+            if (quest != null)
+            {
+                if (Props.refireEveryDays >= 0.0 && quest.State != QuestState.EndedSuccess && quest.State != QuestState.Ongoing && quest.State != QuestState.NotYetAccepted &&
+                    quest.cleanupTick >= 0 && IntervalsPassed == (int) (quest.cleanupTick + Props.refireEveryDays * 60000.0) / 1000)
```
```suggestion
                    quest.cleanupTick >= 0 && IntervalsPassed >= (int) (quest.cleanupTick + Props.refireEveryDays * 60000.0) / 1000)
```

Potential to get stuck I think. If that tick is skipped due to earlier filters intervals passed may be > your "nextFire". I think this should be `>=` instead of `==` but will leave it for you to review just in case I'm wrong and that gets it stuck in some other way.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/HarmonyPatches/ThingStyleHelper_Patch.cs:
```diff
@@ -0,0 +1,21 @@
+    {
+        if (everSeenByPlayer && thing is Thing_ComponentArcho)
+        {
+            Find.World.GetComponent<ArchospringVictoryWorldComponent>().ArchoComponentsSeenByPlayer.Add(thing);
```
potential for duplication; swapping to hashset fixes this

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/Settings.cs:
```diff
@@ -16,16 +18,40 @@ public class Settings : ModSettings
+    public int WaterQuestColonistsAllowed = 10;
+    public int WaterQuestAnimalsAllowed = 10;
+    public int WaterQuestItemsAllowed = 40;
+    public int WaterQuestRelicsAllowed = 10;
```
Pretty sure these are all irrelevant now there's no base move aspect.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/ArchospringVictoryWorldComponent.cs:
```diff
@@ -0,0 +1,30 @@
+{
+    public bool ArchoComponentSeenByPlayer => !ArchoComponentsSeenByPlayer.NullOrEmpty();
+
+    public List<Thing> ArchoComponentsSeenByPlayer = [];
```
should be a hashset

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+
+    public bool Accepts(Thing t)
+    {
+        return (innerContainer.InnerListForReading.Count < 3 || innerContainer.InnerListForReading.Contains(t)) && GetStoreSettings().AllowedToAccept(t) && innerContainer.CanAcceptAnyOf(t);
```
as you've made the amount configurable in settings shouldn't this pull from there instead of hardcoding 3?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+        return (innerContainer.InnerListForReading.Count < 3 || innerContainer.InnerListForReading.Contains(t)) && GetStoreSettings().AllowedToAccept(t) && innerContainer.CanAcceptAnyOf(t);
+    }
+
+    public int SpaceRemainingFor(ThingDef _) => 3 - innerContainer.InnerListForReading.Count;
```
as you've made the amount configurable in settings shouldn't this pull from there instead of hardcoding 3?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+        Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingComponentCount.SetOrAdd(def, innerContainer.Count);
+    }
+
+    public void Notify_ItemRemoved(Thing_ComponentArcho item)
```
shouldn't this remove from the count you added to above?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+    public void Notify_ItemAdded(Thing_ComponentArcho item)
+    {
+        MapHeld.listerHaulables.Notify_AddedThing(item);
+        Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingComponentCount.SetOrAdd(def, innerContainer.Count);
```
with this implementation you can never have more than one of these per def or they'll fight.

keyz182 replied:
There should only ever be one per Def, otherwise something is very wrong.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+    {
+        StringBuilder sb = new StringBuilder(base.GetInspectString());
+
+        sb.Append("MSS_Thirst_BuildingContainsComponents".Translate(innerContainer.Count));
```
perhaps $X/$Y ?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -0,0 +1,133 @@
+
+        Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingAvailable.SetOrAdd(def, false);
+
+        if(innerContainer.Count < 3)
```
as you've made the amount configurable in settings shouldn't this pull from there instead of hardcoding 3?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/MapGen/SitePartWorker_BarbCamp.cs:
```diff
@@ -0,0 +1,36 @@
+    {
+        base.Init(site, sitePart);
+
+        ThingDef compDef = QuestGen.slate.Get<ThingDef>("archospringComponent");
```
I don't see this being added to a slate? Also it should be prefixed

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/MapGen/SitePartWorker_BarbCamp.cs:
```diff
@@ -0,0 +1,36 @@
+        sitePart.things = new ThingOwner<Thing>(sitePart);
+
+        component = ThingMaker.MakeThing(compDef);
+        component = component.MakeMinified();
```
idk why this wouldn't be just one line XD

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/MapGen/SymbolResolver_ArchoBuilding.cs:
```diff
@@ -0,0 +1,129 @@
+                  Log.Error($"Exception thrown while trying to destroy thing: {e}. Continuing");
+              }
+          }
+      }
```
you want to wrap the whole thing in a try finally to make sure that the `Thing.allowDestroyNonDestroyable = false;` definitely fires

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_Cycle.cs:
```diff
@@ -0,0 +1,138 @@
+    protected Map map;
+
+    // Signal name for tracking destruction of the archo buildings in order to reset quests
+    public static string BuildingDestroyedGlobalSignal = "ArchoBuildingDestroyed";
```
Global so prefix?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_Cycle.cs:
```diff
@@ -0,0 +1,138 @@
+      map = QuestGen_Get.GetMap();
+
+      // pre-create any signals we need
+      string activated = QuestGen.GenerateNewSignal("ArchoBuildingComplete");
```
prefix

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_Cycle.cs:
```diff
@@ -0,0 +1,138 @@
+      part1.inSignalDisable = activated;
+      part1.interval = Thirst_Flavour_PackMod.settings.ArchoQuestComponentHuntInterval;
+      part1.archotechComponentDef = Thirst_Flavour_PackDefOf.MSS_Water_ComponentArcho;
+      part1.archotechComponentSlateName = "archotechComponent";
```
shouldn't this be `archospringComponent`? or preferrably `mss_thirst_archospring_component` exxplains why I couldn't find the match earlier

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_FirstCycle.cs:
```diff
@@ -0,0 +1,17 @@
+public class QuestNode_Root_ArchospringVictory_FirstCycle: QuestNode_Root_ArchospringVictory_Cycle
+{
+    protected override int QuestCycle => 1;
+    protected override QuestPart_Activable_ArchoSpringBuilding Activable_ArchoSpringBuilding => new QuestPart_Activable_ArchoSpringBuilding(Thirst_Flavour_PackDefOf.MSS_PowerRegulator, 3);
```
its in the settings no?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_SecondCycle.cs:
```diff
@@ -0,0 +1,30 @@
+public class QuestNode_Root_ArchospringVictory_SecondCycle: QuestNode_Root_ArchospringVictory_Cycle
+{
+    protected override int QuestCycle => 2;
+    protected override QuestPart_Activable_ArchoSpringBuilding Activable_ArchoSpringBuilding => new QuestPart_Activable_ArchoSpringBuilding(Thirst_Flavour_PackDefOf.MSS_CatalyticSeparator, 3);
```
is this not a setting?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_ArchospringVictory_ThirdCycle.cs:
```diff
@@ -0,0 +1,60 @@
+public class QuestNode_Root_ArchospringVictory_ThirdCycle: QuestNode_Root_ArchospringVictory_Cycle
+{
+    protected override int QuestCycle => 3;
+    protected override QuestPart_Activable_ArchoSpringBuilding Activable_ArchoSpringBuilding => new QuestPart_Activable_ArchoSpringBuilding(Thirst_Flavour_PackDefOf.MSS_SterilizationPlant, 3);
```
its in the settings no?

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPartActivable_BuildingUnavailable.cs:
```diff
@@ -0,0 +1,28 @@
+
+    public override void QuestPartTick()
+    {
+        bool val = Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingAvailable.GetWithFallback(BuildingDef, false);
```
only do this after your interval, you don't want to be doing this every tick, it's not huge but it's a for loop not a simple lookup so it's just better form. I'd probably also make this less frequent. 60 is very frequent. 1s on speed 1 is a bit much. Alternatively you can add a static cached getter to the component to avoid the loop but that's probably overkill

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPartActivable_BuildingUnavailable.cs:
```diff
@@ -0,0 +1,28 @@
+    {
+        bool val = Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingAvailable.GetWithFallback(BuildingDef, false);
+
+        if (Find.TickManager.TicksGame % 60 != 0 || val)
```
be a bit careful with this, I don't think quests necessarily tick every tick so you might miss the 60. Better to check less often but keep an int of "nextCheck" tick and if TicksGame < nextCheck then skip.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_Activable_ArchoSpringBuilding.cs:
```diff
@@ -0,0 +1,27 @@
+
+    public override void QuestPartTick()
+    {
+        if (Find.TickManager.TicksGame % 60 != 0 || Find.World.GetComponent<ArchospringVictoryWorldComponent>().BuildingComponentCount.GetWithFallback(Def, 0) < Count)
```
as above, maybe a < instead of !=

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_Filter_ArchoComponent.cs:
```diff
@@ -0,0 +1,35 @@
+
+    public override IEnumerable<GlobalTargetInfo> QuestLookTargets
+    {
+        get => Find.World.GetComponent<ArchospringVictoryWorldComponent>().ArchoComponentsSeenByPlayer.Select(t=>(GlobalTargetInfo)t).Take(3);
```
isn't this a setting

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Rewards/Reward_ArhcospringBuildingSiteMap.cs:
```diff
@@ -0,0 +1,56 @@
+        RulePack customLetterLabelRules,
+        RulePack customLetterTextRules)
+    {
+        throw new NotImplementedException();
```
sus

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Rewards/Reward_ArhcospringBuildingSiteMap.cs:
```diff
@@ -0,0 +1,56 @@
+        RewardsGeneratorParams parms,
+        out float valueActuallyUsed)
+    {
+        throw new NotImplementedException();
```
sus

feldoh commented on Common/Textures/Things/Building/Special/MSS_Water_CatalyticSeparator.png:
no notes; perfection

feldoh commented on Compatibility/sarg.alphapropsparks/Patches/MSS_Thirst_WaterQuest.xml:
```diff
@@ -0,0 +1,90 @@
+        </value>
+    </Operation>
+
+    <!-- <Operation Class="PatchOperationAdd"> -->
```
cruft?

feldoh reviewed (changes requested):
I think there's one or two actual errors I'm gonna push a quick re-prefix which you can review but don't have time to fix the rest today.

feldoh committed:
Re-prefix pass

keyz182 committed:
Remove ignored file

keyz182 committed:
First pass at PR responses

AI Assistant says: Updated global signals for component tracking and introduced a dynamic setting for the required archotech components. Adjusted multiple files for consistency and added proper handling for building destruction and progress checks within the quest system.

keyz182 commented:
I've addressed all but the tick comparisons for now. Those comparisons are from the base game. I'll try to make them more robust, but will do so when I have time to more thoroughly test.

keyz182 committed:
second pass at PR responses - more resilient tick interval checks

keyz182 committed:
Quest to become ally or neutral to get an archo component

keyz182 committed:
Add Teslovs sprites

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestNode_Root_BecomeNeutral.cs:
```diff
@@ -0,0 +1,67 @@
+        FactionRelationKind targetRelationKind = Rand.Bool ? FactionRelationKind.Neutral : FactionRelationKind.Ally;
+        slate.Set("targetRelationKind", targetRelationKind);
+
+        QuestPart_Activalble_FactionRelations questCompleteActivable = new QuestPart_Activalble_FactionRelations(targetFaction, asker.Faction, targetRelationKind);
```
Just a spelling nitpick `QuestPart_Activalble_FactionRelations`
should probably be 
`QuestPart_Activatable_FactionRelations` or `QuestPart_Activable_FactionRelations`

keyz182 replied:
Probably should, but the in-game classes (and the parent class) are `Activable` so stuck with it for consistency

feldoh replied:
Activable not Activalble
you've got a stray `L`

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPartActivable_BuildingUnavailable.cs:
```diff
@@ -10,19 +10,23 @@ namespace Thirst_Flavour_Pack.VictoryQuest;
     {
         base.ExposeData();
         Scribe_Defs.Look(ref BuildingDef, "BuildingDef");
+        Scribe_Values.Look(ref NextCheck, "NextCheck");
```
this is fine but pro tip you don't need to save it. If you're setting `NextCheck = Find.TickManager.TicksGame + 600` then it'll init to 0 so check once at start up then sync itself. This is totally fine so not worth fixing but imo it's a tidier model just because the actual exact tick really doesn't matter

keyz182 replied:
Yeah, I think I got the relationship reversed between the two numbers in my head 🤦

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_Activalble_FactionRelations.cs:
```diff
@@ -0,0 +1,50 @@
+
+        NextCheck += 60;
+
+        if(TargetFaction.PlayerRelationKind == RelationKind)
```
Should this be a better than relationship. i.e. if someone has target neutral but jumps to ally, skipping friends by dumping a load of yayo on them or something I think they'd not complete the quest and they'd have to go and raid them to bring it back down XD

keyz182 replied:
Yup

keyz182 committed:
Cleanup PR issues.

AI says: Fix faction relations typo and remove unused field

Corrected the typo in `QuestPart_Activable_FactionRelations`. Removed the unused field `NextCheck` from `QuestPart_Activable_ArchoSpringBuilding`. Adjusted the `NextCheck` interval and added logic to determine when to complete the quest part based on faction relations.

keyz182 committed:
Final fight work. Not currently working

keyz182 committed:
Final cleanup pass. Working final fight.

AI Says: Refactor Thirst Flavour Pack quest system

Reorganized VictoryQuest classes to improve readability and maintainability. Added a new `QuestPart_Ending` class to handle quest endings. Simplified signal handling and removed unused imports. Enhanced compatibility with added site persistence checks and fixed minor code style issues.

feldoh commented on 1.5/Patches/MSS_Thirst_Archospring.xml:
```diff
@@ -0,0 +1,23 @@
+    <Operation Class="PatchOperationReplace">
+        <xpath>Defs/JobDef[defName="ActivateArchonexusCore"]/reportString</xpath>
+        <value>
+            <reportString>activating archospring.</reportString>
```
should really be a translation :p

keyz182 replied:
Copied from in-game. Not sure it's actually used. Will look to see if it is, and how, and if it's being translated.

keyz182 replied:
looks like none of the reportstrings in-game are transalated that I can see, so leaving for now

feldoh commented on 1.5/Patches/MSS_Thirst_Archospring.xml:
```diff
@@ -0,0 +1,23 @@
+    </Operation>
+
+    <Operation Class="PatchOperationAdd">
+        <xpath>Defs/WorldObjectDef[defName="Site"]/comps</xpath>
```
that doesn't seem right? Won't this add it to ALL sites?

keyz182 replied:
That's how the other Comp for doing detection of all enemies dead is set up. Will see if there's a more targeted way I can do it. Maybe when I create the site, I can add the comp in code?

keyz182 replied:
Wasn't behaving adding it directly, so instead I've just dropped the check interval to every 5 seconds so it's low impact.

feldoh replied:
There's some comp setup shenanigans possibly, what a pity. I wonder if a sitePart would be able to isolate it better 🤷 ah well. bloody weird that the site maker hard codes that one world object type isn't it :(

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/Building_ArchoMachine.cs:
```diff
@@ -26,6 +26,11 @@ public void GetChildHolders(List<IThingHolder> outChildren)
 
+    public bool IsComplete()
+    {
+        return innerContainer.Count >= 3;
```
shouldn't this be using the ArchotechComponentsToCompleteBuilding setting

keyz182 replied:
Yes

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/MapGen/SymbolResolver_Archospring.cs:
```diff
@@ -226,34 +226,33 @@ public override void Resolve(ResolveParams rp)
-          };
-          BaseGen.symbolStack.Push("soundOneShotAction", mechanoidWakeUpParams);
-      }
+      // // Add sleeping mechs
```
cruft

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_Activable_ArchoSpringBuilding.cs:
```diff
@@ -15,7 +15,6 @@ public override void ExposeData()
     {
         base.ExposeData();
         Scribe_Defs.Look(ref Def, "Def");
-        Scribe_Values.Look(ref NextCheck, "NextCheck");
```
dropping this is correct but the NextCheck should be `NextCheck = Find.TickManager.TicksGame + 600;` not `NextCheck += 600;` because otherwise post load it'll run like a thousand times to catch up.

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_Activable_FactionRelations.cs:
```diff
@@ -1,29 +1,33 @@
             return;
 
-        NextCheck += 60;
+        NextCheck += 300;
```
```suggestion
        NextCheck = Find.TickManager.TicksGame + 300;
```

feldoh commented on 1.5/Source/Thirst_Flavour_Pack/VictoryQuest/QuestPart_SubquestGenerator_ArchoHunt.cs:
```diff
@@ -19,7 +19,7 @@ public class QuestPart_SubquestGenerator_ArchoHunt : QuestPart_SubquestGenerator
     public override IEnumerable<GlobalTargetInfo> QuestLookTargets
     {
-        get => Find.World.GetComponent<ArchospringVictoryWorldComponent>().ArchoComponentsSeenByPlayer.Select(t=>(GlobalTargetInfo)t);
+        get => Find.World.GetComponent<ArchospringVictoryWorldComponent>().ArchoComponentsSeenByPlayer.Where(t=>t.ParentHolder is not Building_ArchoMachine).Select(t=>(GlobalTargetInfo)t).Take(3);
```
shouldn't this use the setting

keyz182 replied:
This is just picking a few to show in the UI on the quest page as hyperlinks. I picked 3 as it "looked" the cleanest. More made the quest page crowded. But I can point it to the setting anyway.

keyz182 committed:
Cruft cleanup

keyz182 committed:
Nextcheck tick fix

keyz182 committed:
Use setting for num components

keyz182 committed:
Check for enemies defeated less often

keyz182 committed:
Rebuild

keyz182 committed:
Some debug logging and debug build

keyz182 committed:
Fix quest for full modpack.

AI Says: Remove QuestPart_Activable_ArchoSpringBuilding, refactor related logic

Removed the deprecated QuestPart_Activable_ArchoSpringBuilding class and migrated its functionality to other parts of the code. Refactored signal handling for quest completion, component archomachines, and overall quest cycle management to enhance maintainability and readability.

keyz182 committed:
Rebuild as release

keyz182 committed:
Fix doubled up description for pre-quest

MrSamuelStreamer commented:
Initial merge, will retest
